src/Query.tsx

```diff
@@ -147,86 +184,92 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
 
   private hasMounted: boolean;
   private operation: IDocumentDefinition;
+  static getDerivedStateFromProps(
+    nextProps: InnerQueryProps<any, any>,
+    prevState: QueryState<any>,
+  ) {
+    // if we aren't working from a live query, we can just ignore props changes
+    if (!prevState.queryObservable) return null;
 
-  constructor(props: QueryProps<TData, TVariables>, context: QueryContext) {
-    super(props, context);
+    // if there are no changes to the props, don't do anything state wise
+    if (shallowEqual(nextProps, prevState.props)) return null;
```

Comment:
I'm surprised there wasn't more of a firestorm over the lack of a `prevProps` argument to the `getDerivedStateFromProps` function. (Maybe I missed the firestorm, or maybe people haven't realized yet how ugly it's going to be to have to store props in state strictly for the sake of comparison.)

Either way, stashing the _entire_ prior `props` in `state` already has a code smell. If possible, it'd be good to evaluate the prop-change dependencies of each state key, as opposed to these wholesale comparison short-circuits.

Reply:
More importantly, it looks like this class never updates `state.props` to the new value of `props`. I think every `return` within the method after this line needs to include `{ props: nextProps }`.